sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -1188,6 +1198,36 @@ def _ITE(*args):
     raises(ValueError, lambda: _ITE((a, x < 2), (b, x > 3)))
 
 
+def test_Piecewise_rewrite_as_sign():
+    from itertools import permutations
+
+    def test_perm(args, expected=None):
+        for arg in permutations(args):
+            assert Piecewise(*arg).rewrite(sign) == \
+                expected if expected is not None else Piecewise(*arg).rewrite(sign)
+            assert Piecewise(*arg[:-1], (arg[-1][0], True)).rewrite(sign) == \
+                expected if expected is not None else \
+                Piecewise(*arg[:-1], (arg[-1][0], True)).rewrite(sign)
+
+    mult = lambda f, args: [(f*e, c) for e, c in args]
+    strict_args = [((1, x > 0), (0, Eq(x, 0)), (-1, x < 0))]
+    approx_args = [((1, x >= 0), (-1, x < 0)),
+                   ((1, x > 0), (-1, x <= 0))]
+
+    for args in strict_args:
+        test_perm(args          ,  sign(x))
+        test_perm(mult(-1, args), -sign(x))
+        test_perm(mult( 2, args),  2*sign(x))
+        test_perm(mult( x, args),  x*sign(x))
+        test_perm(mult(-x, args), -x*sign(x))
```

Comment:
This is still quite hard to decipher. For one thing `strict_args` is a list of length one so we don't need a loop.

Also I think that:
```
test_perm([(1, x > 0), (0, Eq(x, 0)), (-1, x < 0)], sign(x))
```
is the most readable way to write this.

Reply:
I put it in a single item list because I imagined he would have more later.